sbnobj/Common/Calibration/TrackCaloSkimmerObj.h

```diff
@@ -1,187 +1,109 @@
 #define SBN_TrackCaloSkimmerObj
 
-#include <climits>
 #include <limits> // for std::numeric_limits
-
+#include <cstdint>
 #include <vector>
 
 namespace sbn {
   struct Vector3D {
-    float x;
-    float y;
-    float z;
-
-    Vector3D():
-      x(std::numeric_limits<float>::signaling_NaN()),
-      y(std::numeric_limits<float>::signaling_NaN()),
-      z(std::numeric_limits<float>::signaling_NaN())
-    {}
-
+    float x = std::numeric_limits<float>::signaling_NaN();
+    float y = std::numeric_limits<float>::signaling_NaN();
+    float z = std::numeric_limits<float>::signaling_NaN();
   };
 
   struct WireInfo {
-    uint16_t wire; //!< Wire number
-    uint16_t plane; //!< Plane number
-    uint16_t tpc; //!< TPC number
-    uint16_t channel; //!< Channel number
-    int16_t tdc0; //!< TDC tick of the first ADC value
-    std::vector<short> adcs; //!< List of ADC values
-
-    WireInfo():
-      wire((uint16_t)-1),
-      plane((uint16_t)-1),
-      tpc((uint16_t)-1),
-      tdc0((uint16_t)-1) {}
-
+    uint16_t wire    = uint16_t(-1); //!< Wire number
+    uint16_t plane   = uint16_t(-1); //!< Plane number
+    uint16_t tpc     = uint16_t(-1); //!< TPC number
+    uint16_t channel = uint16_t(-1); //!< Channel number
+    int16_t tdc0     = int16_t(-1) ; //!< TDC tick of the first ADC value
+    std::vector<short> adcs;         //!< List of ADC values
   };
 
   struct HitTruth {
-    float e;
-    float nelec;
+    float e     = std::numeric_limits<float>::signaling_NaN();
+    float nelec = std::numeric_limits<float>::signaling_NaN();
   };
 
   struct HitInfo {
-    float integral; //!< Integral of gaussian fit to ADC values in hit [ADC]
-    float sumadc; //!< "SummedADC" -- sum of ADC values under gaussian fit [ADC]
-    float width; //!< Width of fitted gaussian hit [ticks]
-    float goodness; //!< Goodness-of-fit of fitted gaussian hit
-    Vector3D sp; //!< Space-Point Position of hit [cm]
-    float time; //!< Peak time of hit [ticks]
-    int id; //!< ID of hit
-    uint16_t channel; //!< Channel number of hit
-    uint16_t wire; //!< Wire number of hit
-    uint16_t plane; //!< Plane number of hit
-    uint16_t tpc; //!< TPC number of hit
-    uint16_t mult; //!< Multiplicity of hit
-    int16_t start; //!< Start tick of hit [ticks]
-    int16_t end; //!< End tick of hit [ticks]
-    bool hasSP; //!< Whether the hit has a SpacePoint
+    float integral = std::numeric_limits<float>::signaling_NaN();    //!< Integral of gaussian fit to ADC values in hit [ADC]
+    float sumadc   = std::numeric_limits<float>::signaling_NaN();    //!< "SummedADC" -- sum of ADC values under gaussian fit [ADC]
+    float width    = std::numeric_limits<float>::signaling_NaN();    //!< Width of fitted gaussian hit [ticks]
+    float goodness = std::numeric_limits<float>::signaling_NaN();    //!< Goodness-of-fit of fitted gaussian hit
+    float time     = std::numeric_limits<float>::signaling_NaN();    //!< Peak time of hit [ticks]
```

Comment:
Note that changing the default values of these data members from `-1` to `NaN` is a **breaking change** (for example, it will change the outcome of the test `if (integral >= 0)`).
Also note, however, that `-1` is not a fool-proof default value for `integral`, `sumadc` and `time` (for this last there is kind of a convention that it will be always positive for any meaningful detector activity, but it should not be relied on).